apps/cmp.c

```diff
@@ -809,7 +809,11 @@ static int load_certs_preliminary(const char *file, STACK_OF(X509) **certs,
     int ret = 0;
 
     if (format == FORMAT_PKCS12) {
-        BIO *bio = bio_open_default(file, 'r', format);
+        FILE *fp = fopen(file, "r");
+        BIO *bio = NULL;
+
+        if (fp != NULL)
+            bio = BIO_new_fp(fp, format);
```

Comment:
If `BIO_new_fp(fp, format)` fails (because `BIO_new(BIO_s_file()` fails) then `fp` needs to be closed.